src/js/hyperschedule.ts

```diff
@@ -854,15 +897,86 @@ function processSearchText() {
       }, filterKeywordsValues)
     ) {
       filtersText.push(text);
+    } else if (isTimeRange(text)) {
+      // Looks only at one time range.
+      timeText = text;
     } else {
       queryText.push(text);
     }
   }
 
   const query = getSearchQuery(queryText);
   const filters = getSearchTextFilters(filtersText);
+  const time = getTimeFilter(timeText);
+  return { query, filters, time };
+}
+
+function isTimeRange(searchText: string) {
+  const timeArray = searchText.split(":");
+  // Time can be singular or a range
+  // Ex: 3:00PM or 3:00PM-4:00PM
+  return (
+    (timeArray.length === 3 &&
+      !isNaN(+timeArray[0]) &&
+      timeArray[1].includes("-")) ||
+    (timeArray.length === 2 &&
+      !isNaN(+timeArray[0].charAt(timeArray[0].length - 1)) &&
+      !isNaN(+timeArray[1].substring(0, 2)))
+  );
+}
 
-  return { query, filters };
+function getTimeRange(rel: string, timeText: string) {
+  let newTimeFilter = [];
+  switch (rel) {
+    case ">":
+    case ">=":
+      newTimeFilter.push(timeText);
+      newTimeFilter.push("23:59");
+      break;
+    case "=":
+      newTimeFilter.push(timeText);
+      break;
+    case "<=":
+    case "<":
+      newTimeFilter.push("00:00");
+      newTimeFilter.push(timeText);
+      break;
+    default:
+      newTimeFilter.push(timeText);
+  }
+  return newTimeFilter;
+}
+
+// Returns an array containing either the start time
+// and end time or the start time alone
+function getTimeFilter(timeText: string) {
+  timeText = timeText.toLowerCase();
+  let inputArr = timeText.split("-");
+  // Single time value (ex: 1:00 PM)
+  if (inputArr.length === 1) {
+    const rel = parseInequality(timeText);
+    inputArr = getTimeRange(rel, timeText.substring(rel.length));
+  }
+  let timeArr = [];
+  for (let time of inputArr) {
+    if (time.substring(time.length - 2) === "am") {
+      // remove "am"
+      time = time.substring(0, time.length - 2);
+      if (time.startsWith("12")) {
+        time = "00:".concat(time.substring(3));
+      }
+    } else if (time.substring(time.length - 2) === "pm") {
+      // remove "pm"
+      time = time.substring(0, time.length - 2);
+      let [hour, min] = time.split(":");
+      if (!time.startsWith("12")) {
+        let hourNum = parseInt(hour) + 12;
```

Comment:
`parseInt` should always be called with `parseInt(n, 10)`; otherwise it tries to do some funky thing where it tries to interpret numbers in other bases, etc.

Thanks, Javascript!

Reply:
Good catch! I noticed there are some other places in the code where parseInt is called without base 10. Just btw.